### DISCUSSION TITLE
Improving names.txt feature

### DISCUSSION THREAD

**Discussion by mortii** [mortii/anki-morphs#44] [Ideas]
> @HQYang1979 It works. Two things you need to do.
> 1. Enable the names.txt option
> 2. After selceting a word as a name you should bury the card. Because it will work with Recalc once you restart anki. The reason > why it's like that is because it uses memozation. Basically the function that reads names.txt only reads it once and stores the result for later.
> 3. Probably the best thing to do is to paste a bunch of names in the file.
> 4. Also @mortii working on the frequency analyzer I realized how fast is to generate a hash_map so I think we should remove the memozation from the functiotn so it works even if the user dosen'nt close anki

_Originally posted by @Vilhelm-Ian in https://github.com/mortii/anki-morphs/issues/41#issuecomment-1811707513_

Caching the file content like we do currently is crucial in my opinion, if we were to load the file and read all the content every single time we extract the morphs from an expression that could end up making recalc extremely slow. Instead of removing the cache we should instead delete/invalidate it when the names.txt file is changed.  I assume the `@functools` library has some way of doing this. Maybe trigger a potential cache invalidation in the `add_name_to_file` function?

Thoughts @Vilhelm-Ian ?

**Comment by Vilhelm-Ian** [comment]
It casches based on the variable it is passed. So a potential solution is to pass a new value when the user right click and adds new name

**Comment by Vilhelm-Ian** [comment]
A simple solution would be to use the text file as a variable for the function that generates the hash set. If the file hasn't changed it will return the cached result

**Comment by mortii** [comment]
According to [this comment](https://stackoverflow.com/questions/37653784/how-do-i-use-cache-clear-on-python-functools-lru-cache/37654201#37654201), we can use:
``` python
create_hash_set_out_of_names().cache_clear()
```
that would be amazing. I'll try it out.

Edit:
it needed to be 
``` python
create_hash_set_out_of_names.cache_clear()
```
and it works perfectly!
I'll push the fix soon.
